service/src/main/java/org/cbioportal/genome_nexus/service/cached/BaseCachedVariantAnnotationFetcher.java

```diff
@@ -0,0 +1,102 @@
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.HttpServerErrorException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public abstract class BaseCachedVariantAnnotationFetcher
+    extends BaseCachedExternalResourceFetcher<VariantAnnotation, VariantAnnotationRepository>
+{
+    private static final Log LOG = LogFactory.getLog(BaseCachedVariantAnnotationFetcher.class);
+
+    protected String collection;
+    protected VariantAnnotationRepository repository;
+    protected Class<VariantAnnotation> type;
+    protected ExternalResourceFetcher<VariantAnnotation> fetcher;
+    protected ResourceTransformer<VariantAnnotation> transformer;
+    protected Integer maxPageSize;
+
+    public BaseCachedVariantAnnotationFetcher(String collection,
+                                             VariantAnnotationRepository repository,
+                                             Class<VariantAnnotation> type,
+                                             ExternalResourceFetcher<VariantAnnotation> fetcher,
+                                             ResourceTransformer<VariantAnnotation> transformer)
+    {
+        super(collection, repository, type, fetcher, transformer, Integer.MAX_VALUE);
+    }
+
+    public BaseCachedVariantAnnotationFetcher(String collection,
+                                             VariantAnnotationRepository repository,
+                                             Class<VariantAnnotation> type,
+                                             ExternalResourceFetcher<VariantAnnotation> fetcher,
+                                             ResourceTransformer<VariantAnnotation> transformer,
+                                             Integer maxPageSize)
+    {
+        super(collection, repository, type, fetcher, transformer, maxPageSize);
+    }
+   
+    @Override
+    protected String extractId(VariantAnnotation instance)
+    {
+        return instance.getVariantId();
+    }
+
+    @Override
+    protected String extractId(DBObject dbObject)
+    {
+        return (String)dbObject.get("input");
+    }
+
+    @Override 
+    public VariantAnnotation fetchAndCache(String id) throws ResourceMappingException
+    {
+        VariantAnnotation variantAnnotation = null;
+        try {
+            variantAnnotation = super.fetchAndCache(id);
+            if (variantAnnotation == null) {
+                variantAnnotation = new VariantAnnotation(id);
+            } else {
+                variantAnnotation.setSuccessfullyAnnotated(true);
+            }
+        } catch (Exception e) {
+            return new VariantAnnotation(id);
+        }
+        return variantAnnotation;
+    }
+    
+    @Override
+    public List<VariantAnnotation> fetchAndCache(List<String> ids) throws ResourceMappingException
+    {
+        Map<String, VariantAnnotation> variantResponse = super.constructFetchedMap(ids); 
```

Comment:
Is there a specific reason to use the `super` keyword here? Why not just `this`?

Reply:
changed!